File: woodroid/src/main/java/me/gilo/woodroid/data/api/ProductVariationAPI.java
package me.gilo.woodroid.data.api;


import me.gilo.woodroid.models.Coupon;
import me.gilo.woodroid.models.OrderNote;
import me.gilo.woodroid.models.Product;
import me.gilo.woodroid.models.Variation;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public interface ProductVariationAPI {

    @Headers("Content-Type: application/json")
    @POST("products/{id}/variations")
    Call<Variation> create(@Path("id") int product_id, @Body Variation body);

    @GET("products/{id}/variations/{variation_id}")
    Call<Variation> view(@Path("id") int product_id, @Path("variation_id") int variation_id);

    @GET("products/{id}/variations")
    Call<List<Variation>> list(@Path("id") int product_id);

    @Headers("Content-Type: application/json")
    @PUT("products/{id}/variations/{variation_id}")
    Call<Variation> update(@Path("id") int product_id, @Path("variation_id") int variation_id, @Body Variation body);

    @DELETE("products/{id}/variations/{variation_id}")
    Call<Variation> delete(@Path("id") int product_id, @Path("variation_id") int variation_id);

    @DELETE("products/{id}/variations/{variation_id}")
    Call<Variation> delete(@Path("id") int product_id, @Path("variation_id") int variation_id, @Query("force") boolean force);

    @Headers("Content-Type: application/json")
    @PUT("products/{id}/variations/{variation_id}")
    Call<Variation> batch(@Path("id") int product_id, @Path("variation_id") int variation_id, @Body Product body);

    @GET("products/{id}/variations")
    Call<List<Variation>> filter(@Path("id") int product_id, @QueryMap Map<String, String> filter);

}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/ProductVariationAPI.kt
package me.gilo.woodroid.data.api


import me.gilo.woodroid.models.Product
import me.gilo.woodroid.models.Variation
import retrofit2.Call
import retrofit2.http.*

interface ProductVariationAPI {

    @Headers("Content-Type: application/json")
    @POST("products/{id}/variations")
    fun create(@Path("id") product_id: Int, @Body body: Variation): Call<Variation>

    @GET("products/{id}/variations/{variation_id}")
    fun view(@Path("id") product_id: Int, @Path("variation_id") variation_id: Int): Call<Variation>

    @GET("products/{id}/variations")
    fun list(@Path("id") product_id: Int): Call<List<Variation>>

    @Headers("Content-Type: application/json")
    @PUT("products/{id}/variations/{variation_id}")
    fun update(@Path("id") product_id: Int, @Path("variation_id") variation_id: Int, @Body body: Variation): Call<Variation>

    @DELETE("products/{id}/variations/{variation_id}")
    fun delete(@Path("id") product_id: Int, @Path("variation_id") variation_id: Int): Call<Variation>

    @DELETE("products/{id}/variations/{variation_id}")
    fun delete(@Path("id") product_id: Int, @Path("variation_id") variation_id: Int, @Query("force") force: Boolean): Call<Variation>

    @Headers("Content-Type: application/json")
    @PUT("products/{id}/variations/{variation_id}")
    fun batch(@Path("id") product_id: Int, @Path("variation_id") variation_id: Int, @Body body: Product): Call<Variation>

    @GET("products/{id}/variations")
    fun filter(@Path("id") product_id: Int, @QueryMap filter: Map<String, String>): Call<List<Variation>>

}